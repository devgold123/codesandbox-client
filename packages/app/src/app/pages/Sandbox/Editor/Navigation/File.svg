<?xml version="1.0" encoding="UTF-8"?>
<svg width="40px" height="40px" viewBox="0 0 40 40" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 48.2 (47327) - http://www.bohemiancoding.com/sketch -->
    <title>Artboard Copy</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Artboard-Copy" fill="#000000" fill-rule="nonzero">
            <g id="Files" transform="translate(4.000000, 0.000000)">
                <path d="M4,0 C1.8,0 0.02,1.8 0.02,4 L0,36 C0,38.2 1.78,40 3.98,40 L28,40 C30.2,40 32,38.2 32,36 L32,12 L20,0 L4,0 Z M18,14 L18,3 L29,14 L18,14 Z" id="Shape"></path>
            </g>
        </g>
    </g>
</svg>