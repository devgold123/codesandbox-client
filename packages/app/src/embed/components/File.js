// @flow

import * as React from 'react';
import styled from 'styled-components';
import Entry from 'app/pages/Sandbox/Editor/Workspace/EntryContainer';
import EntryIcons from 'app/pages/Sandbox/Editor/Workspace/Files/DirectoryEntry/Entry/EntryIcons';
import EntryTitle from 'app/pages/Sandbox/Editor/Workspace/Files/DirectoryEntry/Entry/EntryTitle';

type Props = {
  id: string,
  shortid: string,
  title: string,
  type: 'directory-open' | string,
  setCurrentModule: (shortid: string, id: string) => void,
  depth: number,
  active?: boolean,
  alternative?: boolean,
};
const LeftOffset = styled.div`
  display: flex;
  flex-wrap: nowrap;
  padding-left: ${props => props.depth}rem;
`;

export default class File extends React.PureComponent<Props> {
  static defaultProps = {
    active: false,
    alternative: false,
  };

  setCurrentModule = () => {
    const { id, shortid, setCurrentModule } = this.props;

    setCurrentModule(id, shortid);
  };

  render() {
    const { title, depth, type, active, alternative } = this.props;
    return (
      <div>
        <Entry
          alternative={alternative}
          active={active}
          onClick={this.setCurrentModule}
          type={type}
        >
          <LeftOffset depth={depth}>
            <EntryIcons type={type} />
            <EntryTitle title={title} />
          </LeftOffset>
        </Entry>
      </div>
    );
  }
}
