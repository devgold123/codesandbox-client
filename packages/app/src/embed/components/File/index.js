// @flow
import * as React from 'react';
import { EntryContainer as Entry } from 'app/pages/Sandbox/Editor/Workspace/elements';
import EntryIcons from 'app/pages/Sandbox/Editor/Workspace/Files/DirectoryEntry/Entry/EntryIcons';
import EntryTitle from 'app/pages/Sandbox/Editor/Workspace/Files/DirectoryEntry/Entry/EntryTitle';

import { LeftOffset } from './elements';

type Props = {
  id: string,
  setCurrentModule: (id: string) => void,
  title: string,
  depth: number,
  type: string,
  active?: boolean,
  alternative?: boolean,
};

export default class File extends React.PureComponent<Props> {
  static defaultProps = {
    active: false,
    alternative: false,
  };

  setCurrentModule = () => {
    const { id, setCurrentModule } = this.props;

    setCurrentModule(id);
  };

  render() {
    const { title, depth, type, active, alternative } = this.props;
    return (
      <div>
        <Entry
          alternative={alternative}
          active={active}
          onClick={this.setCurrentModule}
          type={type}
        >
          <LeftOffset depth={depth}>
            <EntryIcons type={type} />
            <EntryTitle title={title} />
          </LeftOffset>
        </Entry>
      </div>
    );
  }
}
