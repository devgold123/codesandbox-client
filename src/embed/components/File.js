// @flow

import React from 'react';
import styled from 'styled-components';
import FunctionIconSvg from 'react-icons/lib/fa/code';
import DirectoryIconSvg from 'react-icons/lib/go/file-directory';
import Entry from 'app/pages/Sandbox/Editor/Workspace/EntryContainer';

type Props = {
  id: string,
  title: 'module' | 'directory',
  type: string,
  setCurrentModule: (id: string) => void,
  depth: number,
  active?: boolean,
  alternative?: boolean,
};
const LeftOffset = styled.div`
  padding-left: ${props => props.depth}rem;
`;

export default class File extends React.PureComponent {
  props: Props;

  getIcon = () => {
    const { type } = this.props;

    const Icon = type === 'directory' ? DirectoryIconSvg : FunctionIconSvg;
    const StyledIcon = styled(Icon)`
      margin-right: 0.5rem;
      vertical-align: middle;
    `;

    return <StyledIcon />;
  };

  setCurrentModule = () => {
    const { id, setCurrentModule } = this.props;

    setCurrentModule(id);
  };

  render() {
    const { title, depth, active, alternative } = this.props;
    return (
      <div>
        <Entry
          alternative={alternative}
          active={active}
          onClick={this.setCurrentModule}
        >
          <LeftOffset depth={depth}>{this.getIcon()} {title}</LeftOffset>
        </Entry>
      </div>
    );
  }
}
