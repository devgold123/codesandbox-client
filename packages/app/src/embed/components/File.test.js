import React from 'react';
import testRender from 'app/utils/test/render';
import File from './File';

describe('File', () => {
  it('renders', () => {
    testRender(<File title="Test" type="module" />);
  });

  it('renders as active', () => {
    testRender(<File active title="Test" type="module" />);
  });

  it('renders directory', () => {
    testRender(<File title="Test" type="directory" />);
  });

  it('renders alternative', () => {
    testRender(<File title="Test" alternative type="directory" />);
  });
});
