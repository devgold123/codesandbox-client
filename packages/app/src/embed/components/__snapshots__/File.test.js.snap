// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File renders 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  padding: 0.4rem;
  padding-left: calc(1rem - 2px);
  padding-right: 3rem;
  color: rgba(255,255,255,0.5);
  text-decoration: none;
  font-weight: 400;
  min-width: 100px;
  border-left: 2px solid transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 function (styleProps) {
  return: styleProps.noTransition || 'transition: 0.3s ease all;';
}

.c0:hover {
  background-color: rgba(108,174,221,0.09999999999999998);
  color: rgb(227,230,232);
  border-color: rgb(38,110,161);
}

.c0:hover > div {
  opacity: 1 !important;
}

.c2 {
  background-image: url(test-file-stub);
  background-size: 1rem;
  background-position: 0;
  background-repeat: no-repeat;
  width: 16px;
  height: 16px;
  display: inline-block;
  -webkit-font-smoothing: antialiased;
  vertical-align: top;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  display: inline-block;
  margin-left: 0.5rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  vertical-align: middle;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-left: rem;
}

<div>
  <span
    className="c0"
    onClick={[Function]}
    type="module"
  >
    <div
      className="c1"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          }
        }
      >
        <span
          className="c2"
          type="module"
        />
      </div>
      <span
        className="c3"
        title="Test"
      >
        Test
      </span>
    </div>
  </span>
</div>
`;

exports[`File renders alternative 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  padding: 0.4rem;
  padding-left: calc(1rem - 2px);
  padding-right: 3rem;
  color: rgba(255,255,255,0.5);
  text-decoration: none;
  font-weight: 400;
  min-width: 100px;
  border-left: 2px solid transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 function (styleProps) {
  return: styleProps.noTransition || 'transition: 0.3s ease all;';
}

.c0:hover {
  background-color: rgba(255,211,153,0.09999999999999998);
  color: rgb(227,230,232);
  border-color: rgb(245,139,0);
}

.c0:hover > div {
  opacity: 1 !important;
}

.c2 {
  background-image: url(test-file-stub);
  background-size: 1rem;
  background-position: 0;
  background-repeat: no-repeat;
  width: 16px;
  height: 16px;
  display: inline-block;
  -webkit-font-smoothing: antialiased;
  vertical-align: top;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  display: inline-block;
  margin-left: 0.5rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  vertical-align: middle;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-left: rem;
}

<div>
  <span
    className="c0"
    onClick={[Function]}
    type="directory"
  >
    <div
      className="c1"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          }
        }
      >
        <span
          className="c2"
          type="directory"
        />
      </div>
      <span
        className="c3"
        title="Test"
      >
        Test
      </span>
    </div>
  </span>
</div>
`;

exports[`File renders as active 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  padding: 0.4rem;
  padding-left: calc(1rem - 2px);
  padding-right: 3rem;
  color: rgba(255,255,255,0.5);
  text-decoration: none;
  font-weight: 400;
  min-width: 100px;
  border-left: 2px solid transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #E0E0E0 !important;
  border-color: #6CAEDD !important;
  background-color: rgba(139,191,228,0.19999999999999996) !important;
}

.c0 function (styleProps) {
  return: styleProps.noTransition || 'transition: 0.3s ease all;';
}

.c0:hover > div {
  opacity: 1 !important;
}

.c2 {
  background-image: url(test-file-stub);
  background-size: 1rem;
  background-position: 0;
  background-repeat: no-repeat;
  width: 16px;
  height: 16px;
  display: inline-block;
  -webkit-font-smoothing: antialiased;
  vertical-align: top;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  display: inline-block;
  margin-left: 0.5rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  vertical-align: middle;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-left: rem;
}

<div>
  <span
    className="c0"
    onClick={[Function]}
    type="module"
  >
    <div
      className="c1"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          }
        }
      >
        <span
          className="c2"
          type="module"
        />
      </div>
      <span
        className="c3"
        title="Test"
      >
        Test
      </span>
    </div>
  </span>
</div>
`;

exports[`File renders directory 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  padding: 0.4rem;
  padding-left: calc(1rem - 2px);
  padding-right: 3rem;
  color: rgba(255,255,255,0.5);
  text-decoration: none;
  font-weight: 400;
  min-width: 100px;
  border-left: 2px solid transparent;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 function (styleProps) {
  return: styleProps.noTransition || 'transition: 0.3s ease all;';
}

.c0:hover {
  background-color: rgba(108,174,221,0.09999999999999998);
  color: rgb(227,230,232);
  border-color: rgb(38,110,161);
}

.c0:hover > div {
  opacity: 1 !important;
}

.c2 {
  background-image: url(test-file-stub);
  background-size: 1rem;
  background-position: 0;
  background-repeat: no-repeat;
  width: 16px;
  height: 16px;
  display: inline-block;
  -webkit-font-smoothing: antialiased;
  vertical-align: top;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  display: inline-block;
  margin-left: 0.5rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  vertical-align: middle;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-left: rem;
}

<div>
  <span
    className="c0"
    onClick={[Function]}
    type="directory"
  >
    <div
      className="c1"
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          }
        }
      >
        <span
          className="c2"
          type="directory"
        />
      </div>
      <span
        className="c3"
        title="Test"
      >
        Test
      </span>
    </div>
  </span>
</div>
`;
